subprojects/core/src/main/java/org/gradle/internal/buildevents/BuildExceptionReporter.java

```diff
@@ -121,7 +122,27 @@ private void renderSingleBuildException(Throwable failure) {
 
         writeFailureDetails(output, details);
 
-        writeGeneralTips(output);
+        if (!hasCompilationCause(failure)) {
+            writeGeneralTips(output);
+        }
+    }
+
+    private static boolean hasCompilationCause(Throwable failure) {
+        Throwable cause = failure.getCause();
+        while (cause != null) {
+            if (hasOnlyCompilationCauses(cause)) {
+                return true;
+            }
+            cause = cause.getCause();
+        }
+        return false;
+    }
+
+    private static boolean hasOnlyCompilationCauses(Throwable cause) {
+        if(cause instanceof DefaultMultiCauseException) {
+            return ((DefaultMultiCauseException) cause).hasOnlyNonGradleSpecifiedCauses();
+        }
+        return false;
```

Comment:
Shouldn't the private methods use a more generic name too (i.e. `hasOnlyNonGradleSpecifiedCause()` and `hasOnlyNonGradleSpecifiedCauses`)?

Reply:
I agree the naming needs an adjustment here.  Maybe `hasNonGradleSpecificCauseInAncestery` or `hasNonGradleSpecificCauseRec` for the `hasCompilationCause`?  Something to indicate it's recursive.